### PR TITLE
Added configuration for switching between Address Hierarchy and simple address fields

### PR DESCRIPTION
**Pull request by vasharma05**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [x] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary
Adding a configuration for switching between using the address hierarchy and simple address fields.

## Screenshots

*None.*

## Related Issue

*None.*

## Other

*None.*

### CODE REVIEW

File: packages/esm-patient-registration-app/src/patient-registration/field/address/address-hierarchy.component.tsx [hunk 1]
```diff
@@ -78,17 +85,27 @@ export const AddressHierarchy: React.FC = () => {
...
+              <ComboInput
+                key={`combo_input_${index}`}
+                name={attributes.name}
+                labelText={t(attributes.name)}
+                id={attributes.name}
+                setSelectedValue={setSelectedValue}
+                selected={selected}
+              />
+            ))
+          : addressLayout.map((attributes, index) => (
```
Comment (ZacButko) [inline packages/esm-patient-registration-app/src/patient-registration/field/address/address-hierarchy.component.tsx:99]: I'm really weirded out that this is using the same variable `addressLayout` for the dynamic one as the static version... Is this really the case?

File: packages/esm-patient-registration-app/src/patient-registration/field/address/address-hierarchy.component.tsx [hunk 2]
```diff
@@ -78,17 +85,27 @@ export const AddressHierarchy: React.FC = () => {
...
-            setSelectedValue={setSelectedValue}
-            selected={selected}
-          />
-        ))}
+        {useAddressHeirarchy
+          ? addressLayout.map((attributes, index) => (
+              <ComboInput
+                key={`combo_input_${index}`}
+                name={attributes.name}
+                labelText={t(attributes.name)}
```
Comment (ZacButko) [inline packages/esm-patient-registration-app/src/patient-registration/field/address/address-hierarchy.component.tsx:93]: This works, but is missing the fallback text for when `eact-i18next` fails. Do you think its reasonable to add this here and for the static version as well?

**Review by ZacButko — APPROVED** [review]
Thanks Vineet. I'm just curious about some things.

File: packages/esm-patient-registration-app/src/patient-registration/field/address/address-hierarchy.component.tsx (hunk 1 above)
Comment (vasharma05) [reply packages/esm-patient-registration-app/src/patient-registration/field/address/address-hierarchy.component.tsx:99]: The address layout provides the fields to be shown on the patient registration page. Hence we are using the same address layout variable/

File: packages/esm-patient-registration-app/src/patient-registration/field/address/address-hierarchy.component.tsx (hunk 2 above)
Comment (vasharma05) [reply packages/esm-patient-registration-app/src/patient-registration/field/address/address-hierarchy.component.tsx:93]: We have defined the translation keys a few lines above this one, hence the keys and values for the address fields are stored in the translation file and is taken for the specific key.

Comment (ZacButko) [reply packages/esm-patient-registration-app/src/patient-registration/field/address/address-hierarchy.component.tsx:93]: Right, but we've seen that translation file be ignored if react-i18next isn't loaded properly. Our site-wide syntax for this as far as i am aware is `t("variableName", "Fallback Text")`. What do you think?

**Pull request merged**